src/entities/schedule.ts

```diff
@@ -16,6 +23,25 @@ export type Schedule = {
   period: SchedulePeriod;
 };
 
+/**
+ * `schedules` が有効であるかどうかを判定する。
+ * 特殊授業の場合は時限が"指定なし"でもよい。
+ * 同一の `Schedule` を指定することはできない。
+ */
+export const isValidSchedules = (schedules: Schedule[]) => {
+  return (
+    !schedules.some(
+      (schedule) =>
+        schedule.module === "指定なし" ||
+        schedule.day === "指定なし" ||
+        (!specialDayJaList.includes(schedule.day as SpecialDayJa) &&
```

Comment:
```suggestion
        (!specialDayJaList.some(day => day === schedule.day) &&
```
個人的にはこの書き方のほうが型安全なので好みです。
ただここでの`as`はそこまで危険性がなく`includes`のほうが直感的なので`as`を避難する意図はありません。
よかったら検討してみて下さい。

Reply:
勉強になります。
ありがとうございます。